### ISSUE TITLE
Resource usage

### ISSUE THREAD

**Issue by marco6** [ReactiveX/RxCpp#502]
I have problems understanding how to use a resource to create a new observable from scratch. The most trivial example I can come out with is:

``` c++
rxcpp::observable<>::scope(
	[]() { return rxcpp::resource<int>(1); },
	[](rxcpp::resource<int> res) { 
		return rxcpp::observable<>::create<int>(
			[res](rxcpp::subscriber<int> sub) {
				sub.on_next(res.get());
				sub.on_completed();
			}
		); 
	}
);
```

In this case, I would get compilation error as _resource::get_ is not const and my lambda is not mutable. Changing the lambda to mutable leads to #473 . Am I misusing _observable::scope_? I could not find many information about its correct usage... 

Still, to make my code work, I could make _mutable_ the callback in observer as proposed in #473, but I'm wondering if it does makes sense in this case. Is there a compelling reason not to make _resource::get()_ const? By looking at the source I can see that it is just calling _get_ on a _shared_ptr_ (which is const already...).

**Comment by kirkshoop** [comment]
yes, `resource::get` should be const. at some point I would like to revisit the design of scope as well
